benchmark/single-source/KeyPathPerformanceTests.swift

```diff
@@ -0,0 +1,1553 @@
+      .element9 / fixedSizeArray100[keyPath: identity(kp9)]
+    fixedSizeArray100.element1 += fixedSizeArray100
+      .element0 + fixedSizeArray100[keyPath: identity(kp4)]
+    fixedSizeArray100.element5 += fixedSizeArray100
+      .element6 - fixedSizeArray100[keyPath: identity(kp2)]
+    fixedSizeArray100.element4 += fixedSizeArray100
+      .element7 * fixedSizeArray100[keyPath: identity(kp2)]
+    fixedSizeArray100.element1 += fixedSizeArray100
+      .element0 / fixedSizeArray100[keyPath: identity(kp6)]
+    fixedSizeArray100.element4 += fixedSizeArray100
+      .element3 + fixedSizeArray100[keyPath: identity(kp9)]
+    fixedSizeArray100.element0 += fixedSizeArray100
+      .element5 - fixedSizeArray100[keyPath: identity(kp8)]
+    fixedSizeArray100.element0 += fixedSizeArray100
+      .element5 * fixedSizeArray100[keyPath: identity(kp7)]
+    fixedSizeArray100.element7 += fixedSizeArray100
+      .element5 / fixedSizeArray100[keyPath: identity(kp9)]
+    fixedSizeArray100.element2 += fixedSizeArray100
+      .element0 + fixedSizeArray100[keyPath: identity(kp6)]
+    fixedSizeArray100.element8 += fixedSizeArray100
+      .element2 - fixedSizeArray100[keyPath: identity(kp2)]
+    fixedSizeArray100.element9 += fixedSizeArray100
+      .element9 * fixedSizeArray100[keyPath: identity(kp2)]
+    fixedSizeArray100.element6 += fixedSizeArray100
+      .element0 / fixedSizeArray100[keyPath: identity(kp4)]
+    fixedSizeArray100.element4 += fixedSizeArray100
+      .element1 + fixedSizeArray100[keyPath: identity(kp7)]
+    fixedSizeArray100.element5 += fixedSizeArray100
+      .element2 - fixedSizeArray100[keyPath: identity(kp7)]
+    fixedSizeArray100.element9 += fixedSizeArray100
+      .element1 * fixedSizeArray100[keyPath: identity(kp5)]
+    fixedSizeArray100.element6 += fixedSizeArray100
+      .element5 / fixedSizeArray100[keyPath: identity(kp3)]
+    fixedSizeArray100.element0 += fixedSizeArray100
+      .element2 + fixedSizeArray100[keyPath: identity(kp2)]
+    fixedSizeArray100.element1 += fixedSizeArray100
+      .element0 - fixedSizeArray100[keyPath: identity(kp1)]
+    fixedSizeArray100.element6 += fixedSizeArray100
+      .element0 * fixedSizeArray100[keyPath: identity(kp7)]
+    fixedSizeArray100.element6 += fixedSizeArray100
+      .element2 / fixedSizeArray100[keyPath: identity(kp0)]
+    fixedSizeArray100.element6 += fixedSizeArray100
+      .element5 + fixedSizeArray100[keyPath: identity(kp0)]
+    fixedSizeArray100.element3 += fixedSizeArray100
+      .element8 - fixedSizeArray100[keyPath: identity(kp0)]
+  }
+  let sum = computeSumOfFixedSizeArray10(fixedSizeArray10: &fixedSizeArray100)
+  blackHole(sum)
+}
+
+// Benchmarks traversal through `KeyPathComponents` of type `.mutatingGetSet`.
+public func run_KeyPathMutatingGetSet(n: Int) {
+  var sum = 0
+  var index = 0
+  let iterationMultipier = 200
+
+  let destinationKeyPath = \MutatingGetSetNested1.nestedItem.nestedItem.nestedItem.nestedItem
```

Comment:
You should wrap the keypath expression into `identity()` to prevent constant folding of the keypath:
```
  let destinationKeyPath = identity(\MutatingGetSetNested1.nestedItem.nestedItem.nestedItem.nestedItem
Write)
```

(in all the benchmark run-functions)